components/inputs/input-number.js

```diff
@@ -69,7 +70,14 @@ class InputNumber extends FormElementMixin(LocalizeCoreElement(LitElement)) {
 		this.requestUpdate('value', oldValue);
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('blur', this._handleBlur);
```

Comment:
Could this be wired up via `<d2l-input-text @blur="{this._handleBlur}">`? That way Lit will automatically remove the listener for you. I see the date input is doing it the same way you are, so maybe it needs to be this way?